### ISSUE TITLE
Blog Post: using slots in Vue with Laravel

### ISSUE THREAD

**Issue by james2doyle** [james2doyle/ohdoylerules.com#29]

**Comment by james2doyle** [comment]
Maybe do WordPress too since its much more crude

**Comment by james2doyle** [comment]
```
When to not use slot?
Single-page apps or isometric javascript (server rendered React/Vue). These cases should contain all their content and not rely on content from other services. With the exception of APIs. But you may choose to call those APIs on the server still. That is how Gatsby and Next/Nuxt render API-driven components to flat HTML. In any case, I don't see a strong reason to use slots when you are working with SPAs and server-rendered JS.
When to use slot?
Server-rendered apps that can't prerender JS components. That would be things like Shopify or PHP. We can avoid complicated and verbose props and try to send down as much rendered HTML as possible. And when the browser is ready, it sprinkles on the JS interactions. This helps keep the site SEO-efficient, feeling super fast, and avoiding flashes of unstyled content (FOUC) while the browser parses all the JS and CSS. So the strategy should be to make sure we can get as much real HTML content on the page as possible before the JS and CSS kicks in. On the server we have access to the page/state "data", but once we need to pass that to Vue, we end up having tons of props with giant JSON strings passed into them. This makes our components much more complex because they will have larger initialization steps, bigger props and data functions, and also need to handle a larger amount of templating inside the component itself. There is a better choice though. We can merge the nice APIs and directives available in Vue, but with the prerendered content from the server. That is where we would use a slot.
A note about the examples - just pretend these are server rendered and the content in the slots came from Liquid or blade, etc.
Simple Accordion: https://codesandbox.io/s/slots-accordion-32c18
This example shows how to use slots to make a simple accordion. The Vue component in this case is purely functional. It contains no styling or hardcoded elements outside of the sections (slots) that need to be filled in. We pass a single function for the accordion buttons back to the "buttons" slot and a single prop for the selected element's index to the "content items" slot.
PDP Image Picker: https://codesandbox.io/s/slots-image-picker-3yvlh
This example shows something with a little more functionality and styling. There is a required prop that sets the default state as well as default attributes that would be rendered on the server and not require javascript to work on page load.
You can actually see these examples are super similar. We are mostly passing a single value (number or string, in this case) back and forth between the slots. Basically, how you might use events with $emit and $on. Same deal, but we dont need Vue to scaffold out all the HTML before we can add the interactions. We skip that step entirely with slots.
```

**Comment by james2doyle** [comment]
Closed with 113122c18cbd5fe17515f2653dcad84e133b594a
